src/xrpld/overlay/detail/OverlayImpl.cpp

```diff
@@ -1199,17 +1200,40 @@ OverlayImpl::getManifestsMessage()
 void
 OverlayImpl::relay(
     uint256 const& hash,
-    protocol::TMTransaction& m,
+    std::optional<std::reference_wrapper<protocol::TMTransaction>> tx,
     std::set<Peer::id_t> const& toSkip)
 {
-    auto const sm = std::make_shared<Message>(m, protocol::mtTRANSACTION);
+    std::shared_ptr<Message> sm = nullptr;
```

Comment:
`sm` is not used before it is set, only set once (a couple dozen lines down from here), and not modified once it's set. Remove this line, and change back to `auto const sm`.
```suggestion
```